runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -289,18 +289,17 @@ function LanguageTree:_get_injections()
       -- You can specify the content and language together
       -- using a tag with the language, for example
       -- @javascript
-      for id, node in pairs(match) do
+      for id, node, data in match:iter() do
         local name = self._injection_query.captures[id]
-        -- TODO add a way to offset the content passed to the parser.
-        -- Needed to shave off leading quotes and things of that nature.
+        local offset_range = data and data.offset
```

Comment:
How would this work if one match would produce more than one range? Will there be more data objects or is there only one data?

Reply:
A match would only have a single injection node, which is where the offset would apply to. The metadata here is for that capture id in the match, so it's specific to that node. The only way it would collide is if you were to add multiple offsets for the same capture and node (which wouldn't really make sense).

```scheme
(pair
  left: ((idenitifier) @left (#offset! @left 0 1 0 -1))
  right: ((identifier) @right (#offset! @right 2 2 2 2)))
```

These would be perfectly fine as they both reference different captures in the match.

Reply:
Thanks for the explanation!

Reply:
You could produce multiple offsets for the same capture I suppose... right now it the last one would overwrite the first. The use case wouldn't apply to injections but I could see it being applicable somewhere else.

Reply:
I'm not wanting to produce multiple results with `offset!`. I'm rather saying that injections might require something that produces multiple ranges.

Would the implementation then be `data and data.offset` then have a single range to insert to the injections and if `data and data.ranges` (or subranges) then insert multiple injections?

Reply:
I see what you are saying. This is the plan for another PR where you the injection can be multiple ranges... This would be another directive (almost like a strip! directive) since the logic would be more complex and a little heavier. Offset is the quicker way of just stripping a known amount of characters and/or lines. So, I see this getting slightly adjusted in the future.